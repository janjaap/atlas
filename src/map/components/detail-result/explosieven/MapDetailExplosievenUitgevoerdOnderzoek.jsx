import React from 'react';
import PropTypes from 'prop-types';

import MapDetailResultDateItem from '../MapDetailResultDateItem';
import MapDetailResultItem from '../MapDetailResultItem';
import MapDetailResultWrapper from '../MapDetailResultWrapper';

const MapDetailExplosievenUitgevoerdOnderzoek = ({
  panoUrl, uitgevoerdOnderzoek, onMaximize, onPanoPreviewClick
}) => (
  <MapDetailResultWrapper
    panoUrl={panoUrl}
    onMaximize={onMaximize}
    onPanoPreviewClick={onPanoPreviewClick}
    subTitle={uitgevoerdOnderzoek.label}
    title="Reeds uitgevoerd CE onderzoek"
  >
    <ul className="map-detail-result__list">
      <MapDetailResultDateItem
        label="Datum rapport"
        date={uitgevoerdOnderzoek.date}
      />
      <MapDetailResultItem
        label="Soort rapportage"
        value={uitgevoerdOnderzoek.type}
      />
      <MapDetailResultItem
        label="Onderzoeksgebied"
        value={uitgevoerdOnderzoek.onderzoeksgebied}
      />
      <MapDetailResultItem
        label="Verdacht gebied"
        value={uitgevoerdOnderzoek.verdachtGebied}
      />
    </ul>
  </MapDetailResultWrapper>
);

MapDetailExplosievenUitgevoerdOnderzoek.propTypes = {
  uitgevoerdOnderzoek: PropTypes.shape({
    date: PropTypes.instanceOf(Date),
    label: PropTypes.string,
    onderzoeksgebied: PropTypes.string,
    type: PropTypes.string,
    verdachtGebied: PropTypes.string
  }).isRequired,
  panoUrl: PropTypes.string.isRequired,
  onMaximize: PropTypes.func.isRequired,
  onPanoPreviewClick: PropTypes.func.isRequired
};

export default MapDetailExplosievenUitgevoerdOnderzoek;
