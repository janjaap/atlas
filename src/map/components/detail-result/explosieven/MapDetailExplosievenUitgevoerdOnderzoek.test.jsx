import React from 'react';
import { shallow } from 'enzyme';

import MapDetailExplosievenUitgevoerdOnderzoek from './MapDetailExplosievenUitgevoerdOnderzoek';

describe('MapDetailExplosievenUitgevoerdOnderzoek', () => {
  it('should render everything', () => {
    const uitgevoerdOnderzoek = {
      date: new Date('1919-12-31'),
      label: 'Uitgevoerd onderzoek label',
      onderzoeksgebied: 'Uitgevoerd onderzoek onderzoeksgebied',
      type: 'Uitgevoerd onderzoek type',
      verdachtGebied: 'Uitgevoerd onderzoek verdacht gebied'
    };
    const clickHandler = jest.fn();
    const wrapper = shallow(
      <MapDetailExplosievenUitgevoerdOnderzoek
        panoUrl="panoUrl"
        onMaximize={clickHandler}
        onPanoPreviewClick={clickHandler}
        uitgevoerdOnderzoek={uitgevoerdOnderzoek}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
