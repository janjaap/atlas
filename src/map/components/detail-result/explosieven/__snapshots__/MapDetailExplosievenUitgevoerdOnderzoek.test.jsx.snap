// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapDetailExplosievenUitgevoerdOnderzoek should render everything 1`] = `
<MapDetailResultWrapper
  onMaximize={[MockFunction]}
  onPanoPreviewClick={[MockFunction]}
  panoUrl="panoUrl"
  subTitle="Uitgevoerd onderzoek label"
  title="Reeds uitgevoerd CE onderzoek"
>
  <ul
    className="map-detail-result__list"
  >
    <MapDetailResultDateItem
      date={1919-12-31T00:00:00.000Z}
      label="Datum rapport"
    />
    <MapDetailResultItem
      label="Soort rapportage"
      value="Uitgevoerd onderzoek type"
    />
    <MapDetailResultItem
      label="Onderzoeksgebied"
      value="Uitgevoerd onderzoek onderzoeksgebied"
    />
    <MapDetailResultItem
      label="Verdacht gebied"
      value="Uitgevoerd onderzoek verdacht gebied"
    />
  </ul>
</MapDetailResultWrapper>
`;
